fsgrid.hpp

```diff
@@ -1040,3 +1086,108 @@ template <typename T, int stencil> class FsGrid : public FsGridTools{
          array[2] = a;
       }
 };
+
+
+//Array Operator Overloads
+template <typename T, size_t N>
+static inline std::array<T,N> operator+(const std::array<T, N>& arr1, const std::array<T, N>& arr2){
+   std::array<T, N> res;
+    for (size_t i = 0; i < N; ++i){
+        res[i] = arr1[i] + arr2[i];
+      }
+    return res; 
+}
+
+template <typename T, size_t N>
+static inline std::array<T,N> operator-(const std::array<T, N>& arr1, const std::array<T, N>& arr2){
+   std::array<T, N> res;
+    for (size_t i = 0; i < N; ++i){
+        res[i] = arr1[i] - arr2[i];
+      }
+    return res; 
+}
+
+template <typename T, size_t N>
+static inline std::array<T, N> operator*(const std::array<T, N> &arr1, const std::array<T, N> &arr2){
+   std::array<T, N> res;
+   for (size_t i = 0; i < N; ++i){
+      res[i] = arr1[i] * arr2[i];
+   }
+   return res;
+}
+
+template <typename T, size_t N>
+static inline std::array<T,N> operator/(const std::array<T, N>& arr1, const std::array<T, N>& arr2){
+   std::array<T, N> res;
+    for (size_t i = 0; i < N; ++i){
+        res[i] = arr1[i] / arr2[i];
+      }
+    return res; 
+}
+
+template <typename T, size_t N,typename S,typename std::enable_if<std::is_arithmetic<S>::value>::type* = nullptr>
+std::array<T,N> operator*(std::array<T, N>& arr, S mult){
+    for (size_t i = 0; i < N; ++i)
+        arr[i]*=mult;
+    return arr; 
+}
+
+
+
+// FsGrid Operator Overloads
+template <class T, int stencil>
+static inline FsGrid<T, stencil> operator+(const FsGrid<T, stencil> &lhs, const FsGrid<T, stencil> &rhs){
+   FsGrid<T, stencil> a(lhs);
+   return a+=rhs;
+}
+
+template <class T, int stencil>
+static inline FsGrid<T, stencil> operator-(const FsGrid<T, stencil> &lhs, const FsGrid<T, stencil> &rhs){
+   FsGrid<T, stencil> a(lhs);
+   return a-=rhs;
+}
+
+template <class T, int stencil>
+static inline FsGrid<T, stencil> operator*(const FsGrid<T, stencil> &lhs, const FsGrid<T, stencil> &rhs){
+   FsGrid<T, stencil> a(lhs);
+   return a *= rhs;
+}
+
+template <class T, int stencil>
+static inline FsGrid<T, stencil> operator/(const FsGrid<T, stencil> &lhs, const FsGrid<T, stencil> &rhs){
+   FsGrid<T, stencil> a(lhs);
+   return a /= rhs;
+}
+
+template <class T, int stencil,typename S,typename std::enable_if<std::is_arithmetic<S>::value>::type* = nullptr>
+static inline FsGrid<T, stencil>  lerp_t(const  FsGrid<T, stencil> &lhs, const FsGrid<T, stencil> &rhs,S t0, S t1,S t){
+   
+   //Sanity check for timestamps. If sizes mismatch then this will make it crash in the '=' operator
+   bool doAble = (t0<=t) && (t<=t1);
+   if (!doAble){
+      #pragma message(": warning<FsGrid lerp_t>: We should probably crash here instead of returning nonesense")
+      //Do we crash here?
+      std::cerr<<"Interpolation not doable in the range of "<<t0<<" "<<t<<" "<<t1<<std::endl; 
+      return lhs;
+   }
```

Comment:
Not sure if this should even be an error. If we are outside the [t0,t1] range, this simply yields a linear extrapolation. Nothing wrong about that.

Reply:
for the new fieldsolver testing, I in fact want to be able to extrapolate beyond 1! :)

Reply:
Roger that @markusbattarbee. I am pushing an update with some new access operators and that will also be able to extrapolate as well